admin/services/pupil.status.service.js

```diff
@@ -117,11 +117,14 @@ pupilStatusService.recalculateStatusByPupilSlugs = async (pupilSlugs, schoolId)
  * @param {number} schoolId - ID of school in school table
  */
 pupilStatusService.recalculateStatusByPupilIds = async (pupilIds, schoolId) => {
-  if (!(Array.isArray(pupilIds) && pupilIds.length > 0)) {
+  // After exceeding 20 items the request payload received contains object key-value pairs
+  // Detecting and converting them to strings is necessary as part of the processing
+  const processedPupilIds = pupilIds.map(p => typeof p === 'object' ? Object.values(p)[0] : p)
```

Comment:
I would prefer it if it was the controller's job to translate from the incoming web request to the format (Array) required by the service layer.

Reply:
@jon-shipley this is a data processing activity. I thought the same thing but don't we have services to cover that part?
Shouldn't controllers just be as much as logic free as possible?

Reply:
@GeorgeChatzigiannis 

It is explicitly the controller's job to translate from the IO device (the web) in order to call the business tier.  I don't see why we *want* the business tier to handle Arrays and Objects.

Reply:
@jon-shipley changes made